### PR TITLE
P20-394: Create unit tests for Dashboard blocks i18n sync-out

### PR DESCRIPTION
**Pull request by artem-vavilov**
1. Modularized the Dashboard `blocks` resource `sync-out` business logic
3. Optimized  the Dashboard `blocks` resource `sync-in` business logic
4. Updated the Dashboard `blocks` resource `sync-in` unit tests
6. Created unit tests for the Dashboard `blocks` resource `sync-out` business logic
8. Created executable script `bundle exec bin/i18n/resources/dashboard/blocks/sync_in.rb`
9. Created executable script `bundle exec bin/i18n/resources/dashboard/blocks/sync_out.rb`

## Links
- jira ticket: [P20-394](https://codedotorg.atlassian.net/browse/P20-394)

### CODE REVIEW

File: bin/i18n/i18n_script_utils.rb [hunk 1]
```diff
@@ -378,6 +387,16 @@ def self.copy_file(file_path, dest_path)
...
+  # Moves file
+  #
+  # @param file_path [String] path to the file
+  # @param dest_path [String] destination path
+  def self.move_file(file_path, dest_path)
+    dest_dir = File.extname(dest_path).empty? ? dest_path : File.dirname(dest_path)
+    FileUtils.mkdir_p(dest_dir)
+    FileUtils.mv(file_path, dest_path, force: true)
+  end
+
```
Comment (mgc1194) [inline bin/i18n/i18n_script_utils.rb:399]: I see tests for `write_file` and `copy_file`, but not for `move_file`.

Comment (artem-vavilov) [reply bin/i18n/i18n_script_utils.rb:399]: https://github.com/code-dot-org/code-dot-org/blob/83f0924e27f42dd3d2c9fec72db07551f9c3c6c0/bin/test/i18n/test_i18n_script_utils.rb#L338

File: bin/i18n/i18n_script_utils.rb [hunk 2]
```diff
@@ -378,10 +387,20 @@ def self.copy_file(file_path, dest_path)
...
   # Renames directory
   #
-  # @param [String] from_dir, e.g. `i18n/locales/English/resource`
-  # @param [String] to_dir, e.g. `i18n/locales/en-US/resource`
+  # @param from_dir [String] the dir origin path, e.g. `i18n/locales/English/resource`
```
Comment (mgc1194) [inline bin/i18n/i18n_script_utils.rb]:
```suggestion
  # @param from_dir [String] the original directory path name, e.g. `i18n/locales/English/resource`
```

File: bin/i18n/i18n_script_utils.rb [hunk 3]
```diff
@@ -378,10 +387,20 @@ def self.copy_file(file_path, dest_path)
...
+  # @param to_dir [String] the dir new path, e.g. `i18n/locales/en-US/resource`
```
Comment (mgc1194) [inline bin/i18n/i18n_script_utils.rb]:
```suggestion
  # @param to_dir [String] the new directory path name, e.g. `i18n/locales/en-US/resource`
```

**Review by mgc1194 — APPROVED** [review]
I ran the sync and got files synced out.

**Pull request merged**